### PR TITLE
gitoxide 0.25.0 (new formula)

### PR DESCRIPTION
**Pull request by georgettica**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by alebcay** [comment]
For the error you encountered, it is complaining about not finding `pkg-config`. You can add `pkg-config` and any other dependencies (e.g. some Rust-based formulae need OpenSSL on Linux) [using the `on_linux` block](https://github.com/Homebrew/homebrew-core/blob/cd2f827afa5b61a8c41666c326641db467377b43/Formula/lychee.rb#L22-L24).

For future reference, you can `depends_on :macos` or `depends_on :linux` if you need to exclude one OS or another ([for instance, `unxip` uses this to build on macOS only](https://github.com/Homebrew/homebrew-core/blob/cd2f827afa5b61a8c41666c326641db467377b43/Formula/unxip.rb#L18)).

But for formulae submitted to core, it is preferred that it works on both macOS/Linux unless there's some good reason for it not to.

File: Formula/gitoxide.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test gitoxide`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```
Comment (carlocab) [inline Formula/gitoxide.rb:25]:
```suggestion
```

File: Formula/gitoxide.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    assert_match "gix-plumbing", shell_output("#{bin}/gix --version")
```
Comment (carlocab) [inline Formula/gitoxide.rb:26]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/gitoxide.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class Gitoxide < Formula
+  desc "Idiomatic, lean, fast & safe pure Rust implementation of Git"
+  homepage "https://github.com/Byron/gitoxide"
+  url "https://github.com/Byron/gitoxide/archive/refs/tags/v0.25.0.tar.gz"
+  sha256 "098bb18e1cae42ab7597b6b442538d3f51b57935a848ea121e20e2921d6a4693"
+  license "Apache-2.0"
+
+  # Cmake is here as only rust install wasn't enough
```
Comment (carlocab) [inline Formula/gitoxide.rb:8]: Why wasn't it enough?

File: Formula/gitoxide.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "cmake" => :build
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/gitoxide.rb:10]:
```suggestion
  depends_on "cmake" => :build
  depends_on "pkg-config" :build
  depends_on "rust" => :build
```

**Comment by carlocab** [comment]
> For the error you encountered, it is complaining about not finding `pkg-config`. You can add `pkg-config` and any other dependencies (e.g. some Rust-based formulae need OpenSSL on Linux) [using the `on_linux` block](https://github.com/Homebrew/homebrew-core/blob/cd2f827afa5b61a8c41666c326641db467377b43/Formula/lychee.rb#L22-L24).

`pkg-config` is a runtime dependency of `rust`, and is always on `PATH` for macOS builds (because of the `pkg-config` shim), so we should just skip the `on_linux` block for a `pkg-config` dependency when there is already a `rust` dependency.

**Comment by alebcay** [comment]
Adjusted my original comment to reflect that `pkg-config` doesn't need to be enclosed in `on_linux`.

File: Formula/gitoxide.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  url "https://github.com/Byron/gitoxide/archive/refs/tags/v0.25.0.tar.gz"
+  sha256 "098bb18e1cae42ab7597b6b442538d3f51b57935a848ea121e20e2921d6a4693"
+  license "Apache-2.0"
+
+  # Cmake is here as only rust install wasn't enough (explained in PR)
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  # might need to add a "git" test dependency, but locally this fails on
+  # Error: gitoxide is missing test dependencies: git
```
Comment (carlocab) [inline Formula/gitoxide.rb:13]:
```suggestion
  uses_from_macos "curl"
```

File: Formula/gitoxide.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match "gix-plumbing", shell_output("#{bin}/gix --version")
+    system "git", "init", "test", "--quiet"
+    touch "test/file.txt"
+    system "git", "-C", "test", "add", "."
+    system "git", "-C", "test", "commit", "--message", "\"initial commit\"", "--quiet"
```
Comment (carlocab) [inline Formula/gitoxide.rb:24]:
```suggestion
    system "git", "-C", "test", "commit", "--message", "initial commit", "--quiet"
```

File: Formula/gitoxide.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  desc "Idiomatic, lean, fast & safe pure Rust implementation of Git"
+  homepage "https://github.com/Byron/gitoxide"
+  url "https://github.com/Byron/gitoxide/archive/refs/tags/v0.25.0.tar.gz"
+  sha256 "098bb18e1cae42ab7597b6b442538d3f51b57935a848ea121e20e2921d6a4693"
+  license "Apache-2.0"
+
+  # Cmake is here as only rust install wasn't enough (explained in PR)
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/gitoxide.rb:11]:
```suggestion
  depends_on "rust" => :build
  uses_from_macos "curl"
```
You need this to fix the linkage failure on Linux.

File: Formula/gitoxide.rb (hunk 3 above)
Comment (carlocab) [reply Formula/gitoxide.rb]: Pushed a change that I _think_ will allow us to get rid of CMake and use system Zlib instead.

**Review by carlocab — APPROVED** [review]
Thanks!

**Review by p-linnane — APPROVED** [review]

**Pull request merged**